lib/tree.rb

```diff
@@ -1,38 +1,61 @@
 class NoApplesError < StandardError; end
 
+
 class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+  attr_accessor :height, :age, :apples, :alive
 
   def initialize
+    @height = 0
+    @age = 0
+    @apples = []
+    @alive = true
+    @colors = ['red', 'green', 'yellow']
   end
 
   def age!
+    @age += 1
+    @height += rand(1..(2*@age))
+    self.add_apples
   end
 
   def add_apples
+    rand(1..(4*@age)).times do
+      @apples << Apple.new(@colors[rand(0..2)], rand(1..5))
+    end
   end
 
   def any_apples?
+    @apples.length > 0 ? true : false
```

Comment:
There is `empty?`, and at least when you have ActiveSupport available, `present?`.